File: algorithm/src/main/java/baekjoon/p01159/Main.java
package baekjoon.p01159;

import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

public class Main {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        String[] inputs = new String[n];

        for (int i = 0; i < inputs.length; i++) {
            inputs[i] = sc.next();
        }

        List<Character> solves = solve(inputs);
        if (solves.size() == 0) {
            System.out.print("PREDAJA");
            return;
        }

        for (char solve : solves) {
            System.out.print(solve);
        }
    }

    private static List<Character> solve(String[] inputs) {
        int[] arr = new int[26];
        for (String input : inputs) {
            char ch = input.charAt(0);
            arr[ch - 'a']++;
        }

        List<Character> ret = new ArrayList<>();
        for (int i = 0; i < arr.length; i++) {
            if (arr[i] >= 5) {
                ret.add((char) (i + 'a'));
            }
        }

        return ret;
    }

}

File: algorithm/src/main/java/baekjoon/p01159/Main.kt
package baekjoon.p01159

fun main(args: Array<String>) {
    val n = readLine()!!.toInt()
    val inputs = Array(n) { readLine()!! }
    val solves = solve(inputs)

    if (solves.isEmpty()) {
        println("PREDAJA")
    } else {
        println(solves.joinToString(""))
    }
}

private fun solve(inputs: Array<String>): List<Char> {
    val cnt = ('a'..'z').map { inputs.count { input -> (input.first() == it) } }
    val ret = ArrayList<Char>()

    for (i in 0 until cnt.size) {
        if (cnt[i] >= 5) {
            val ch = (i + 'a'.toInt()).toChar()
            ret.add(ch)
        }
    }

    return ret
}